nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/resources/docs/org.apache.nifi.processors.standard.QueryRecord/additionalDetails.html

```diff
@@ -30,19 +31,525 @@
+                        "name": "Apache Software Foundation"
+                   },
+                  "debutYear": 2014
+              }
+            }
+        </code></pre>
+
+        <p>
+            Consider a query that will select the title and name of any person who has a home address in a different state
+            than their work address. Here, we can only select the fields <code>name</code>, <code>title</code>,
+            <code>age</code>, and <code>addresses</code>. In this scenario, <code>addresses</code> represents an Array of complex
+            objects - records. In order to accommodate for this, QueryRecord provides User-Defined Functions to enable
+            <a href="../../../../../html/record-path-guide.html">Record Path</a> to be used. Record Path is a simple NiFi Domain Specific Language (DSL)
+            that allows users to reference a nested structure.
+        </p>
+
+        <p>
+            The primary User-Defined Function that will be used is named <code>RPATH</code> (short for Record Path). This function expects exactly two arguments:
+            the Record to evaluate the RecordPath against, and the RecordPath to evaluate (in that order).
+            So, to select the title and name of any person who has a home address in a different state than their work address, we can use
+            the following SQL statement:
+        </p>
+
+        <code><pre>
+            SELECT title, name
+            FROM FLOWFILE
+            WHERE RPATH(addresses, '/state[/label = ''home'']') <>
+                  RPATH(addresses, '/state[/label = ''work'']')
+        </pre></code>
+
+        <p>
+            To explain this query in English, we can say that it selects the "title" and "name" fields from any Record in the FlowFile for which there is an address whose "label" is "home" and
+            another address whose "label" is "work" and for which the two addreses have different states.
+        </p>
+
+        <p>
+            Similarly, we could select the entire Record (all fields) of any person who has a "project" whose maintainer is the Apache Software Foundation using the query:
+        </p>
+
+        <code><pre>
+            SELECT *
+            FROM FLOWFILE
+            WHERE RPATH(project, '/maintainer/name') = 'Apache Software Foundation'
+        </pre></code>
+
+        <p>
+            There does exist a caveat, though, when using RecordPath. That is that the <code>RPATH</code> function returns an <code>Object</code>, which in JDBC is represented as an <code>OTHER</code>
+            type. This is fine and does not affect anything when it is used like above. However, what if we wanted to use another SQL function on the result? For example, what if we wanted to use
+            the SQL query <code>SELECT * FROM FLOWFILE WHERE RPATH(project, '/maintainer/name') LIKE 'Apache%'</code>? This would fail with a very long error such as:
+        </p>
+
+        <code><pre>
```

Comment:
These docs are great, thanks! However I don't think we need that super-large error message in the user documentation, IMO documenting that you will get a super-large error in those cases should be enough.

Another thought, is it possible to coerce the various types to Strings (as necessary) so Calcite _could_ work with the statement above?

Reply:
I can appreciate the concern, that it may prove to be rather verbose for user documentation. However, in this case, it is a well-known issue that can occur and produces a well-known, verbose error message, but that error message is not very helpful in the given context. In this case, if this very verbose error message does appear, we know what causes it and how to resolve it. Therefore, I do believe it is worth documenting, even given its verbosity. Documenting that a super-large error will occur may not suffice, as we do not know if some other condition could result in an alternate super-large error, which could lead the user astray and leave him/her frustrated. When in doubt, I would lean toward erring on the side of clarity over brevity.

Reply:
Re: coercion: I don't think we have all necessary information to determine that we need to perform the coercion before such an Exception occurs. I could be wrong, but I don't see a clean way to avoid it...